### PR TITLE
Add a blog post with some of the caveats and praise for Delta Lake

### PR DESCRIPTION
**Pull request by rtyler**
@houqp @xianwill I would appreciate a review before 10:15 PDT so I can get this published before my session at Spark and AI Summit on the topic.

### CODE REVIEW

File: _posts/2020-06-26-streaming-with-delta-lake.md [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+important to be aware of when designing a system. Many of our lessons learned
+came by partnering with [Databricks](https://databricks.com) during the design
+phase of the View Analytics project. Fortunately they were able to identify
+some ways in which our original designs would have failed before we ended up
+building and deploying anything to production.
+
+Some things to keep in mind:
+
+* Multiple streams can _append_ to the same table concurrently, *but* if there
+  are any non-append writers (e.g. [merge writers](https://docs.delta.io/latest/delta-update.html)) then no other
```
Comment (houqp) [inline _posts/2020-06-26-streaming-with-delta-lake.md:114]: i am not sure if this is ture, delta's acid transaction log should support any kind of concurrent writes with databricks' proprietary delta locking feature.

Comment (houqp) [reply _posts/2020-06-26-streaming-with-delta-lake.md:114]: i think what we should mention is to only use append only delta table as streaming source.

File: _posts/2020-06-26-streaming-with-delta-lake.md [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+* When there if there are any non-append writers, an optimize cannot run externally. In essence it **must** be executed inline in a streaming job when the merge writer is not running, i.e. periodically within a `foreachBatch`
+* [Checkpoints](https://spark.apache.org/docs/latest/streaming-programming-guide.html#checkpointing) must be managed *carefully*. each checkpoint location should belong exclusively to a single write stream. restarts of the job must always use the same checkpoint location. do not reference the same checkpoint location from multiple write streams as they will overwrite each others checkpoints (very bad).
+
+
+### Optimize
+
+Building further upon the foundation laid by transactions, Delta Lake provides
+an `OPTIMIZE` command, which helps prevent the small files problem entirely.
+
+In the streaming context, it is highly unlikely that event datawill come in
```
Comment (houqp) [inline _posts/2020-06-26-streaming-with-delta-lake.md:125]: split datawill

File: _posts/2020-06-26-streaming-with-delta-lake.md [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+make this a much safer operation when deploying on eventually consistent data
+stores like S3. Delta tables can however also act as a **source** for another streaming consumer.
+
+In Scribd's deployment, this allows us to have some Spark Streaming jobs which
+are consuming directly from [Apache Kafka](https://kafka.apache.org), while
+other downstream streaming jobs consume _from Delta tables themselves_.
+
+My [previous post](/blog/2020/shipping-rust-to-production.html) alluded to the "View Analytics" project, which relies heavily on Delta Lake's streaming support:
+
+![View Analytics data pipeline](/post-images/2020-06-delta-lake/view-analytics.png)
```
Comment (houqp) [inline _posts/2020-06-26-streaming-with-delta-lake.md:91]: i couldn't comment on the image, but there should be a pointer from optimize job to view_logs table as well.

File: _posts/2020-06-26-streaming-with-delta-lake.md [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+phase of the View Analytics project. Fortunately they were able to identify
+some ways in which our original designs would have failed before we ended up
+building and deploying anything to production.
+
+Some things to keep in mind:
+
+* Multiple streams can _append_ to the same table concurrently, *but* if there
+  are any non-append writers (e.g. [merge writers](https://docs.delta.io/latest/delta-update.html)) then no other
+  writers should run concurrently with the non-append writer.
+* When there if there are any non-append writers, an optimize cannot run externally. In essence it **must** be executed inline in a streaming job when the merge writer is not running, i.e. periodically within a `foreachBatch`
```
Comment (houqp) [inline _posts/2020-06-26-streaming-with-delta-lake.md:116]: typo: `When there if there are any non-append writers`. similar to my comment above, i feel like delta's transaction log with databricks' proprietary locking should support running optimize with concurrent upsert wrtiers as well. But I can see it not being a good idea to run optimize externally for a busy streaming job due to more frequent transaction commit conflicts.

File: _posts/2020-06-26-streaming-with-delta-lake.md (hunk 3 above)
Comment (rtyler) [reply _posts/2020-06-26-streaming-with-delta-lake.md:91]: I'm not concerned about 100% accuracy, I just grabbed the same diagram from my Spark Summit presentation :shrug:

**Comment by rtyler** [comment]
Thanks for the review @houqp, I've tried to fix a bit of phrasing to make some of the Deatabricks runtime caveats more clean

**Pull request merged**